### DISCUSSION TITLE
updateIfExists pull request?

### DISCUSSION THREAD

**Discussion by Kondamon** [pointfreeco/swift-identified-collections#61] [Ideas]
Hey all,

thanks for the great framework. I would be interested in having an update function only if an object already exists in the array. Otherwise the update does nothing. If you think this is a useful addition I would create a merge request for this feature.

`public mutating func updateIfExists(_ item: Element) -> Element?`

**Comment by alinless** [comment]
It would be great to have this option!

**Comment by stephencelis** [comment]
I believe you can use optional-chaining to qualify subscript assignment:

```swift
array[id: element.id]? = element
```

Does this work for you?

**Reply by alinless** [reply]
Yes, that works! Didn't know about this feature. Thank you!
